cmd/fs/helpers.go

```diff
@@ -99,30 +93,42 @@ func getValidArgsFunction(pathArgCount int, onlyDirs bool, filerForPathFunc func
 
 		completions, directive := completer.CompleteRemotePath(path)
 
-		// The completions will start with a "/", so we'll prefix
-		// them with the selectedPrefix without a trailing "/"
-		prefix := selectedPrefix[:len(selectedPrefix)-1]
-
-		// Add the prefix to the completions
 		for i, completion := range completions {
-			completions[i] = fmt.Sprintf("%s%s", prefix, completion)
+			if isDbfsPath {
+				// The completions will start with a "/", so we'll prefix them with the
+				// selectedPrefix without a trailing "/"
+				prefix := dbfsPrefix[:len(dbfsPrefix)-1]
+				completions[i] = fmt.Sprintf("%s%s", prefix, completion)
+			} else if shouldDropLocalPrefix(toComplete, completion) {
+
+				completions[i] = completion[len(localPefix):]
+			} else {
+				completions[i] = completion
+			}
+		}
+
+		// If the path is a dbfs path, we try to add the dbfs:/Volumes prefix suggestion
+		if isDbfsPath && strings.HasPrefix(volumesPefix, toComplete) {
+			completions = append([]string{highlight(volumesPefix)}, completions...)
+
 		}
 
-		// If the path is a dbfs path, we should also add the Volumes prefix
-		if strings.HasPrefix(volumesPefix, toComplete) {
-			completions = append(completions, volumesPefix)
+		// If the path is local, we try to prepend the dbfs:/ prefix suggestion
+		if !isDbfsPath && strings.HasPrefix(dbfsPrefix, toComplete) {
+			completions = append([]string{highlight(dbfsPrefix)}, completions...)
 		}
 
 		return completions, directive
 	}
 }
 
-// Wrapper for [root.MustWorkspaceClient] that disables loading authentication configuration from a bundle.
-func mustWorkspaceClient(cmd *cobra.Command, args []string) error {
-	if root.HasWorkspaceClient(cmd.Context()) {
-		return nil
-	}
+// Drop the local prefix from completions if the path to complete doesn't
+// start with it. We do this because the local filer returns paths with the
+// local prefix.
+func shouldDropLocalPrefix(toComplete string, completion string) bool {
+	return !strings.HasPrefix(toComplete, localPefix) && strings.HasPrefix(completion, localPefix)
```

Comment:
but path could have also `../` right? Also we need to take care of Windows-like paths.
Could it be an absolute local path?

Reply:
We don't need to drop `../` path, Windows paths or absolute paths as they're handled correctly:
<img width="484" alt="Screenshot 2024-07-29 at 11 51 53" src="https://github.com/user-attachments/assets/d586285c-b751-41dd-a869-50580b8f9d67">

We only need to drop `./` in cases where the user has typed `foo` and the local filer returns `./foo`. I updated to the comment.

Note: I still have to test this change on a Window VM today.

Reply:
Tested this on windows and it does seem to work as expected:
<img width="751" alt="Screenshot 2024-08-02 at 9 33 19" src="https://github.com/user-attachments/assets/a4a359cf-fcf3-4c49-840f-c751d4eecfff">
<img width="713" alt="Screenshot 2024-08-02 at 9 33 07" src="https://github.com/user-attachments/assets/6e9d2ca1-a017-4fd4-9765-4c4fa8f878cc">

Reply:
Added a `handleLocalPrefix` function that handles both Windows and Unix.